code/modules/mob/living/basic/lavaland/mook/mook_ai.dm

```diff
@@ -0,0 +1,315 @@
+///commands the chief can pick from
+GLOBAL_LIST_INIT(mook_commands, list(
+	new /datum/pet_command/point_targetting/attack(),
+	new /datum/pet_command/point_targetting/fetch(),
+))
+
+/datum/ai_controller/basic_controller/mook
+	blackboard = list(
+		BB_TARGETTING_DATUM = new /datum/targetting_datum/basic/mook,
+		BB_BLACKLIST_MINERAL_TURFS = list(/turf/closed/mineral/gibtonite, /turf/closed/mineral/strong),
+		BB_MAXIMUM_DISTANCE_TO_VILLAGE = 7,
+		BB_STORM_APPROACHING = FALSE,
+	)
+
+	ai_movement = /datum/ai_movement/basic_avoidance
+	idle_behavior = /datum/idle_behavior/idle_random_walk
+	planning_subtrees = list(
+		/datum/ai_planning_subtree/simple_find_target,
+		/datum/ai_planning_subtree/targeted_mob_ability/leap,
+		/datum/ai_planning_subtree/basic_melee_attack_subtree,
+		/datum/ai_planning_subtree/find_and_hunt_target/material_stand,
+		/datum/ai_planning_subtree/use_mob_ability/mook_jump,
+		/datum/ai_planning_subtree/find_and_hunt_target/hunt_ores/mook,
+		/datum/ai_planning_subtree/mine_walls/mook,
+		/datum/ai_planning_subtree/wander_away_from_village,
+	)
+
+///check for faction if not a ash walker, otherwise just attack
+/datum/targetting_datum/basic/mook/faction_check(mob/living/living_mob, mob/living/the_target)
+	if(isashwalker(the_target))
+		return FALSE
+
+	return ..()
+
+/datum/ai_planning_subtree/targeted_mob_ability/leap
+	ability_key = BB_MOOK_LEAP_ABILITY
+
+/datum/ai_planning_subtree/use_mob_ability/mook_jump
+	ability_key = BB_MOOK_JUMP_ABILITY
+
+///jump towards the village when we have found ore or there is a storm coming
+/datum/ai_planning_subtree/use_mob_ability/mook_jump/SelectBehaviors(datum/ai_controller/controller, seconds_per_tick)
+	var/storm_approaching = controller.blackboard[BB_STORM_APPROACHING]
+	var/mob/living/living_pawn = controller.pawn
+	var/obj/effect/home = controller.blackboard[BB_HOME_VILLAGE]
+	if(QDELETED(home))
+		return
+	if(get_dist(living_pawn, home) < controller.blackboard[BB_MAXIMUM_DISTANCE_TO_VILLAGE])
+		return
+	if(storm_approaching || (locate(/obj/item/stack/ore) in living_pawn))
+		controller.clear_blackboard_key(BB_TARGET_MINERAL_WALL)
+		return ..()
```

Comment:
make this an early return instead for clarity because i always think it's weird for a "conditional" call of parent

```suggestion
	if(!storm_approaching || !(locate(/obj/item/stack/ore) in living_pawn))
		return

	controller.clear_blackboard_key(BB_TARGET_MINERAL_WALL)
	return ..()
```

takes up some more space but feels way more logical